segment/writer.go

```diff
@@ -367,15 +367,46 @@ func (w *Writer) appendEntry(e types.LogEntry) error {
 		typ: FrameEntry,
 		len: uint32(len(e.Data)),
 	}
-	return w.appendFrame(fh, e.Data)
+
+	bufOffset, err := w.appendFrame(fh, e.Data)
+	if err != nil {
+		return err
+	}
+
+	// If frame is an entry, update the index
```

Comment:
nit: The if part of this comment doesn't make sense any more since that's always the case in this method.